frameworks/kafka/docs/security.md

```diff
@@ -0,0 +1,242 @@
+```
+<service-primary>/kafka-Y-broker.<service-subdomain>.autoip.dcos.thisdcos.directory@<service-realm>
+```
+with:
+- service-primary = service.security.kerberos.primary
+- Y = 0 up to kafka.brokers.count - 1
+- service-subdomain = service.name with all `/`'s removed
+- service-realm = service.security.kerberos.realm
+
+For example, if installing with these options:
+```
+{
+    "service": {
+        "name": "a/good/example",
+        "kerberos": {
+            "primary": "example",
+            "realm": "EXAMPLE"
+        }
+    },
+    "kafka": {
+        "brokers": {
+            "count": 3
+        }
+    }
+}
+```
+then the principals to create would be:
+```
+example/kafka-0-broker.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/kafka-1-broker.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/kafka-2-broker.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+```
+
+#### Place Service Keytab in DC/OS Secret Store
+
+The DC/OS Apache Kafka service uses a keytab containing all broker principals (service keytab) to simplify orchestration. After creating the principals above, generate the service keytab making sure to include all the broker principals. This will be stored as a secret in the DC/OS Secret Store.
+
+*Note*: DC/OS 1.10 does not support directly adding binary secrets to the secret store, only text files are supported. Instead, first base64 encode the file, and save it to the secret store as `/desired/path/__base_64__secret_name`. The DC/OS security modules will handle decoding the file when it is used by the service. More details [here](https://docs.mesosphere.com/services/ops-guide/overview/#binary-secrets).
+
+The service keytab should be stored at `service/path/service.keytab` (as noted above for 1.10, it would be `__base_64__service.keytab`), where `service/path` matches the path of the service. For example, if installing with the options
+```
+{
+    "service": {
+        "name": "a/good/example"
+    }
+}
+```
+then the service keytab should be stored at `a/good/service.keytab`.
+
+Documentation for adding a file to the secret store can be found [here](https://docs.mesosphere.com/latest/security/ent/secrets/create-secrets/#creating-secrets-from-a-file-via-the-dcos-enterprise-cli).
+
+*Note*: Secrets access is controlled by [DC/OS Spaces](https://docs.mesosphere.com/1.10/security/ent/#spaces-for-secrets). Technically, any secret path "above" that of the service will be accessible by the service. Matching the two paths is, however, the most secure option. Additionally `service.keytab` is a convention and not a requirement.
+
+
+#### Install the service
+
+Install the DC/OS Apache Kafka service will the following options alongside your own:
+```
+{
+    "service": {
+        "kerberos": {
+            "enabled": true,
+            "enabled_for_zookeeper": <true|false default false>
```

Comment:
Missing `,` (adding the `json` tag as mentioned above would help with highlighting this in VSCode)